### DISCUSSION TITLE
File I/O in pybrick EV3 mindstorm

### DISCUSSION THREAD

**Discussion by pekyeen** [pybricks/support#1071] [Q&A] [answered]
Hi, Does anyone know how to use the uio module in pybrick to read a txt file that saved at the same path as the project (main.py)? Normally we do f = open('myfile.txt', 'r') in python. May I know how to use it in pybrick? Appreciate for your response. Thanks!

**Comment by dlech** [comment]
> f = open('myfile.txt', 'r')

This works in Pybricks on ev3dev too.

**Reply by pekyeen** [reply]
Thanks for your reply.. The code is as below

```python
print('start to read file')
f = open("storelocation_bycolor.txt", "r")
print(f.read())
```

but I get this error:
```
start to read file
Traceback (most recent call last):
  File "/home/robot/pybrick_test/main.py", line 30, in <module>
OSError: [Errno 2] ENOENT
----------
Exited with error code 1.
```

Do you have any more description of this error?

**Reply by dlech** [reply]
Can you give the exact steps to reproduce the problem? How did you get the program on the EV3? Did the .txt file also get copied to the EV3? How did you run the program on the EV3? Can you share the full program?

**Comment by pekyeen** [comment, marked as answer]
Hi dlech, thank you so much. I have figured out the issue. The file path location should be like this '../storelocation_bycolor.txt'. Issue resolved. Thank you!
